src/peft/tuners/lora.py

```diff
@@ -271,6 +271,9 @@ def _create_new_module(self, lora_config, adapter_name, target):
                 }
             )
             new_module = Linear4bit(adapter_name, target.in_features, target.out_features, bias=bias, **fourbit_kwargs)
+        elif is_auto_gptq_available() and target.__class__.__name__ == "QuantLinear":
+            new_module = QuantLinear(adapter_name, target, **kwargs)
+            target.weight = target.qweight
```

Comment:
```suggestion
```

Reply:
I didn't do it at first because I don't wanted to import the library as it also uses peft. Let me know if my changes works with you. I needed to protect the import and get the right class depending on the quantization config.